### PR TITLE
Rename sha3 to keccak

### LINKED ISSUE #22: Rename sha3 to keccak256
https://github.com/ethereum/EIPs/issues/59

### PR DESCRIPTION
**Pull request by axic**
Fixes #22.

### CODE REVIEW

**Comment by holgerd77** [comment]
Rebased this.

File: index.js [hunk 1]
```diff
@@ -20,40 +20,46 @@ exports.MAX_INTEGER = new BN('ffffffffffffffffffffffffffffffffffffffffffffffffff
...
 
 /**
- * SHA3-256 hash of the RLP of null (a ```Buffer```)
- * @var {Buffer} SHA3_RLP
+ * Keccak-256 hash of the RLP of null (a ```Buffer```)
+ * @var {Buffer} KECCAK256_RLP
  */
-exports.SHA3_RLP = Buffer.from(exports.SHA3_RLP_S, 'hex')
+exports.KECCAK256_RLP = Buffer.from(exports.KECCAK256_RLP_S, 'hex')
+exports.SHA3_RLP = exports.KECCAK256_RLP
```
Comment (holgerd77) [inline index.js]: The above constant exports look good.

File: index.js [hunk 2]
```diff
@@ -215,18 +221,26 @@ exports.toUnsigned = function (num) {
...
   return createKeccakHash('keccak' + bits).update(a).digest()
 }
 
+/**
+ * Creates SHA-3 (Keccak) hash of the input [OBSOLETE]
+ * @param {Buffer|Array|String|Number} a the input data
+ * @param {Number} [bits=256] the SHA-3 width
+ * @return {Buffer}
+ */
+exports.sha3 = exports.keccak
```
Comment (holgerd77) [inline index.js]: Ok, just renaming to ``keccak`` since there is also the possibility to provide ``bits`` as parameter.

File: index.js [hunk 3]
```diff
@@ -259,7 +273,7 @@ exports.ripemd160 = function (a, padded) {
  * @return {Buffer}
  */
 exports.rlphash = function (a) {
-  return exports.sha3(rlp.encode(a))
+  return exports.keccak(rlp.encode(a))
 }
```
Comment (holgerd77) [inline index.js]: Ok.

File: index.js [hunk 4]
```diff
@@ -358,7 +372,7 @@ exports.ecsign = function (msgHash, privateKey) {
  */
 exports.hashPersonalMessage = function (message) {
   const prefix = exports.toBuffer('\u0019Ethereum Signed Message:\n' + message.length.toString())
-  return exports.sha3(Buffer.concat([prefix, message]))
+  return exports.keccak(Buffer.concat([prefix, message]))
```
Comment (holgerd77) [inline index.js]: Both ok.

File: index.js [hunk 5]
```diff
@@ -464,7 +478,7 @@ exports.isZeroAddress = function (address) {
  */
 exports.toChecksumAddress = function (address) {
   address = exports.stripHexPrefix(address).toLowerCase()
-  const hash = exports.sha3(address).toString('hex')
+  const hash = exports.keccak(address).toString('hex')
```
Comment (holgerd77) [inline index.js]: Ok. Also checked that there is nothing forgotten in ``index.js`` by pre-scrolling the ``master`` branch version to the next position where I would have inserted/changed something.

File: test/index.js [hunk 6]
```diff
@@ -82,7 +82,7 @@ describe('ripemd160', function () {
 })
 
 describe('rlphash', function () {
-  it('should produce a sha3 of the rlp data', function () {
+  it('should produce a keccak-256 hash of the rlp data', function () {
```
Comment (holgerd77) [inline test/index.js]: Ok, also checked for completeness.

File: docs/index.md [hunk 7]
```diff
@@ -471,108 +485,108 @@ Checks if a given address is a zero address
...
+[index.js:97-101](https://github.com/ethereumjs/ethereumjs-util/blob/fde15dd0cc65885dc8731d1bd2facbba78fb39ac/index.js#L97-L101 "Source code on GitHub")
 
 Returns a zero address
 
 Returns **[String](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/String)** 
 
 ## zeros
 
-[index.js:82-84](https://github.com/ethereumjs/ethereumjs-util/blob/0bc0616c2de0de0723f87c3b64923bb9f962a8e1/index.js#L82-L84 "Source code on GitHub")
+[index.js:88-90](https://github.com/ethereumjs/ethereumjs-util/blob/fde15dd0cc65885dc8731d1bd2facbba78fb39ac/index.js#L88-L90 "Source code on GitHub")
```
Comment (holgerd77) [inline docs/index.md]: Had a rough look over the documentation and did a short markdown preview, looks good.

**Comment by axic** [comment]
@holgerd77 should we merge this or wait for #122 ?

**Comment by holgerd77** [comment]
No, no need to wait, will merge.

**Pull request merged**